### DISCUSSION TITLE
Angled xAxis is getting Overflow

### DISCUSSION THREAD

**Discussion by jiaulislam** [recharts/recharts#4211] [Q&A] [answered]
Hi, Sorry if I somehow duplicated the question but I really searched before posting here.  as given in below image the gray color is the div area and dark blue is the recharts ResponsiveContainer area but after doing the xAxis date angle={-30} on the my ticks are getting overflow. what can I do to solve the issue. 

Note: I'm using in Next.js 14, recharts=2.12.0

```javascript
export default function BarChartVerticalGrouped() {


  return (
    <ResponsiveContainer width="100%" height={600} className={"bg-gray-800"}>
      <BarChart
        height={400}
        width={600}
        data={data}
        margin={{
          top: 5,
          right: 30,
          left: 20,
          bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray={"5 5"} />
        <XAxis dataKey={"xLabel"} angle={-30} textAnchor="end" />
        {/* <YAxis dataKey={"turnoverTarget"} /> */}
        <Tooltip />
        <Legend verticalAlign="top" height={36}/>
        <Bar
          dataKey={"turnoverGenerated"}
          fill="#8884d8"
          activeBar={<Rectangle fill="pink" stroke="blue" />}
        />
        <Bar
          dataKey={"turnoverTarget"}
          fill="#82ca9d"
          activeBar={<Rectangle fill="gold" stroke="purple" />}
        />
      </BarChart>
    </ResponsiveContainer>
  );
}
```

![image](https://github.com/recharts/recharts/assets/54698236/1184ee58-0c8b-4266-b74e-cc30c4cd3ded)

**Comment by ckifer** [comment, marked as answer]
Looks like you just need to add more margin bottom, it doesn't get adjusted automatically

**Reply by jiaulislam** [reply]
Well ans was the margin but I had to set height for the xAxis. Then the issue was solved
